portal/src/util/types.ts

```diff
@@ -13,3 +14,12 @@ export function isPortalApiApp(value: any): value is PortalAPIApp {
   }
   return value.__typename === "App";
 }
+
+export function isUserDetails(
```

Comment:
No need to have this type guard. The generated type should be correct already. Please also remove `isPortalApiApp` above.

Reply:
without typeguard, typescript cannot decide the type of the node. In authentication screen query, there is two possible type, `__typename = "User" | "App"`, As the node implements both User and App. If we need to cast it to the correct type we will need `as any as ...` ? @louischan-oursky

Reply:
I see. Let's move this function to the file where the query is defined then. It is weird that `/src/util/types.ts` imports a generated file of a specific component.